dcos_launch/aws.py

```diff
@@ -85,18 +85,21 @@ def describe(self):
 
     def delete(self):
         self.stack.delete()
-        if len(self.config['temp_resources']) > 0:
-            # must wait for stack to be deleted before removing
-            # network resources on which it depends
-            self.stack.wait_for_complete(
-                transition_states=[
-                    'CREATE_COMPLETE',
-                    'UPDATE_COMPLETE_CLEANUP_IN_PROGRESS',
-                    'UPDATE_IN_PROGRESS',
-                    'UPDATE_COMPLETE',
-                    'DELETE_IN_PROGRESS'
-                ],
-                end_states=['DELETE_COMPLETE'])
+        # we must wait for the stack to be deleted for 2 reasons:
+        # 1. required to remove network resources on which it depends
+        # 2. to make sure it successfully deletes. If not (for example got interrupted by tagging), we retry deleting
+        status = self.stack.wait_for_complete(
+            transition_states=[
+                'CREATE_COMPLETE',
+                'UPDATE_COMPLETE_CLEANUP_IN_PROGRESS',
+                'UPDATE_IN_PROGRESS',
+                'DELETE_IN_PROGRESS'
+            ],
+            end_states=['DELETE_COMPLETE', 'UPDATE_COMPLETE'])
+        if status == 'UPDATE_COMPLETE':
```

Comment:
Can you explain via comment as why this should be the case and how did we determine this?  That is state changed to "UPDATE_COMPLETE" implies deletion was interrupted?

Reply:
The comment for it is a couple of lines above https://github.com/dcos/dcos-launch/pull/129/files/5a1a328f6d41f8352654d63421e153907c53a5de#diff-338b729a3fc00f84d642b1018a052e2eR90